### DISCUSSION TITLE
Problem with Plugin Installation

### DISCUSSION THREAD

**Discussion by maxzapp** [netbox-community/netbox#17702] [Q&A]
I installed on a fresh Ubuntu 24.04.1 python 3.12 Netbox v4.1.3 with gunicorn and nginx as a proof of concept according to the official documentation .
I can access Netbox login and so on, so I assume the install seems to be correct.

As I need some essential plugins I try to install them . I am aware that plugin development may be behind the actual netbox version. But I cannot install a single one. With pip install <package>I always got a
„ERROR: could not install packages due to an OSError 13 Permission denied /opt/netbox-4.1.3/venv/lib/python3.12/site-packages/netbox_reorder_rack. Check permissions 

Of course the whole app under /opt/netbox is owned by root.

If  I try to sudo pip install….   I got another error: externally-managed-environment

So what did I wrong?

**Comment by a084ed22** [comment]
Put the package in local_requirements.txt and rerun the script. Your venv is likely with improper permissions set, therefore the user netbox runs as can't add packages. You should also not install packages system-wide, but within the venv.

**Reply by maxzapp** [reply]
Hi, thx for your tip. If I enter the plugins in local_requirements and in configuration.py and run ./upgrade.py the plugins are loaded.

All folders and files in /opt/netbox. are owned by root is this not correct?

**Reply by a084ed22** [reply]
I believe it is correct for most files in /opt/netbox to be owned by root. Only the media subfolder within /opt/netbox/netbox should be writable by the netbox process and in general, to allow uploading of content.

Depending on how the installation was performed, your issue was likely a mismatch between the user with which you installed the plugin and the general netbox installation.
